README.md

```diff
@@ -1,33 +1,222 @@
+1. Product - value assigned in item or variation level.
+
+All properties defined in parent hierarchy entities are available in (inherited by) item. E.g. every item inside catalog will have property “Brand” with value “Sony” once it was defined in parent catalog and was assigned such value there.
+
+Every property has its “Assign to” trait set to one of: “catalog”, “category”, “product” or “variation”. This defines at what level the property’s value can be set.
+
+![Fig. Properties](docs/media/screen-product-properties.png)
+
+#### Scenarios
+
+[View Properties](/docs/view-properties.md)
+
+[Manage Properties](/docs/manage-properties.md)
+
+### Linked Category
+
+Common category has the same meaning as the category in common catalogs. User can create common category and link products to it or create linked category in it.
+
+catalog CRUD:
+
+- properties
+
+category CRUD:
+
+- properties;
+- SEO.
+
+virtual catalog CRUD:
+
+- categories;
+- including categories from other (real) catalogs.
+
+Expandability points:
+
+- additional widgets in (virtual) catalog detail;
+- additional widgets in category detail;
+- additional widgets in item detail.
+
+## VirtoCommerce Search
+
+The solution uses Search as primary tool for customer interaction. The customers expect high-level relevance, multiple-languages and near-instantaneous responses.
+
+### Benefits
+
+1. Document based full text search and text analysis for Products, Categories and Members;
+
+1. Search by keyword, filters, aggregations and geo-search;
+
+1. Supports multiple search engines: Elastic Search, Elastic Cloud, Azure Search and Lucence;
+
+1. Functionality is exposed through REST API or .NET SDK that makes the inherent complexity of information retrieval;
+
+1. The support documents can be easily extended by third party team.
+
+### Architecture Overview
+
+The Architecture is focused to isolate the search engine from documents and provide abilities to add new types of documents.
+
+[Search Engine Architecture Details](/docs/search-engine-architecture-details.md)
+
+[Catalog Search](/docs/catalog-search.md)
+
+### Elastic Search
+
+https://github.com/VirtoCommerce/vc-module-elastic-search
+
+### Azure Search
+
+https://github.com/VirtoCommerce/vc-module-azure-search
+
+### Lucene Search
+
+https://github.com/VirtoCommerce/vc-module-lucene-search
+
+This Search Module is used only for development.
+
+### Search Provider Tests
+
+https://github.com/VirtoCommerce/vc-module-core/blob/dev/VirtoCommerce.CoreModule.Search.Tests/SearchProviderTests.cs
+
+## Search Index
+
+VirtoCommerce platform provides a unified solution for indexing and search of different types of objects. The search engine is unable to operate directly with the database, especially when full text search, aggregations, partial search are needed. For this reason, from the architecture point of view, it was decided to isolate the search engine from the documents that should be searched.
```

Comment:
There is  a significant difference between 'unable' and 'shouldn't' !! What is the correct definition from the technical point of view? 'Cos it's said 'unable' in Oleg's presentation.

Reply:
Because of that difference I think it should be "shouldn't". Search engine could be implemented with direct database usage, but it would be inefficient, so it should not be implemented that way. Let's ask @OlegoO opinion.

Reply:
we are still waiting for this point to be clarified